vignettes/parse_lookup.Rmd

```diff
@@ -0,0 +1,127 @@
+
+## `parse()`
+---
+
+The `parse()` function is intended to extract identifiable drug names from a corpus of text such as, clinical trial data, social media, survey or interview transcription. `parse()` takes in one argument, the vector that contains the strings to be parsed.
+
+Here is an example of some problematic records in the `drug_df` dataset that warrants the use of `parse()`
+
+```{r messy}
+
+drug_df %>% 
+  filter(str_detect(textdrug, ",|;|and|\\/|=|\\(")) %>% 
+  distinct(textdrug)
+
+```
+
+As you can see there are so many extraneous/problematic characters, multiple drugs in one record and several variations of the same drug (i.e. "bup/nx"). We assume that the user is solely interested in the drugs themselves, not information such as dosage and units. 
+
+This messy data is exactly what `parse()` was designed for. 
+
+```{r}
+drug_names <- parse(drug_df$textdrug)
+
+head(drug_names)
+```
+
+The resulting vector can then be passed on to the `lookup_*` functions to identify whether the input drug is a class or category or a synonym for other drugs in the same category. 
+
+## `lookup_*`
+---
+
+### `lookup()`
+
+This function relies on a comprehensive lookup table `lookup_df`. This dataframe contains 3 variables:  
+- `class` =  
+- `category` =  
+- `synonym` =  
+
+```{r lookup_df, echo=TRUE}
+dim(lookup_df)
+
+str(lookup_df)
+
+class(lookup_df)
+```
+
+The purpose of this function is to return any possible matches to the lookup_df, which is a comprehensive dataframe consisting of all drug classes, categories and synonyms. It serves as a source or helper function to many of the other more specific function in the package. The idea is to match any possible columns with a the single word, a list of separate words or a vector passed as an argument. The dataframe returned will consist of the lookup_df match as well as the `original_word` that was the source of the match. 
+
+Here is an example of a common search done using `lookup`. 
+
+```{r}
+results <- lookup("crack", "pizza", "marijuana")
+head(results)
+```
+
+You can see that the dataframe returned could be vast in it's matches, and that the other more specific functions, below, might be of more use depending on one's needs. 
+
+---
+
+### `compress_lookup()`
+
+This function takes in one argument: the table returned from a search using the `lookup` function. The purpose of this function is to narrow down the results to a more specific dataframe consisting of only relevant values, such as `Class` and/or `Category` depending on the users selection. However, it defaults to returning just `Class`, `Category` and `Original_Word` from the large list of matches. 
+
+
+If a researcher wanted to determine the main classes of drugs being used by the patients of a clinical study, they might pass a large vector of words from clinical notes taken in a study to the `lookup` function, then filter them down to only the return the datafram of classes relevant to their needs.
```

Comment:
* i wouldn't say "large vector of words" - since lookup_df only contains drugs of abuse. maybe "large vector of substances"?
* extra "the" after "filter them down"